pkg/addon/template.go

```diff
@@ -189,6 +189,10 @@ func ParseFromFolder(log *zap.SugaredLogger, overwriteRegistry string, manifestP
 
 		// recurse into subdirectory
 		if info.IsDir() {
+			if info.Name() == "aws-node-termination-handler" && data.Cluster.CloudProviderName != "aws" {
+				continue
+			}
```

Comment:
We usually handle this by checking `cloudProviderName` in the YAML manifest. For example: 

https://github.com/kubermatic/kubermatic/blob/e5d9543c577ade55a4aebfcf38fefcbdde1464a3/addons/csi/hetzner/hcloud-csi.yaml#L21

Is there a specific reason why we don't use this approach here?

Reply:
hmm not really, this also could work. And I would use it instead of this check. Thanks